### PR TITLE
Link index to borrower list in ui 

### PR DESCRIPTION
**Pull request by hoholyin**
- For commands `return` and `renew`, the given parameter `INDEX` now refers to the borrower's loaned list instead of the main catalog list.

- Fixed bug in serial number generator to generate the correct next serial number

- Added restriction to book title and author (30 char max) so as to accommodate to UI and loan slip

- Removed irrelevant tests in `RenewCommandTest` and `ReturnCommandTest` (eg. Returning a book that is not loaned by the borrower is not possible when using the return command with index from the borrower's own book)

### CODE REVIEW

File: src/test/java/seedu/address/model/book/BookTest.java [hunk 1]
```diff
@@ -15,10 +14,6 @@
 
 import org.junit.jupiter.api.Test;
 
-//import seedu.address.commons.util.DateUtil;
-//import seedu.address.model.borrower.BorrowerId;
-//import seedu.address.model.loan.Loan;
```
Comment (seeziyang) [inline src/test/java/seedu/address/model/book/BookTest.java:20]: Are these imports still needed?

File: src/main/java/seedu/address/logic/commands/ReturnCommand.java [hunk 2]
```diff
@@ -57,12 +57,12 @@ public CommandResult execute(Model model) throws CommandException {
...
 
-        List<Book> lastShownList = model.getFilteredBookList();
-        if (index.getZeroBased() >= lastShownList.size()) {
+        List<Book> lastShownBorrowerBooksList = model.getBorrowerBooks();
+        if (index.getZeroBased() >= lastShownBorrowerBooksList.size()) {
             throw new CommandException(MESSAGE_INVALID_BOOK_DISPLAYED_INDEX);
         }
 
-        Book bookToBeReturned = lastShownList.get(index.getZeroBased()); // TODO change to second list index
+        Book bookToBeReturned = lastShownBorrowerBooksList.get(index.getZeroBased());
```
Comment (seeziyang) [inline src/main/java/seedu/address/logic/commands/ReturnCommand.java:65]: Thanks!

**Review by seeziyang — APPROVED** [review]
LGTM!

File: src/test/java/seedu/address/model/book/BookTest.java (hunk 1 above)
Comment (hoholyin) [reply src/test/java/seedu/address/model/book/BookTest.java:20]: Can add them back when we need them, none of the tests using them at the moment

**Comment by seeziyang** [comment]
I will merge first and try to fix that test case.

**Pull request merged**